sklearn/metrics/_classification.py

```diff
@@ -360,6 +376,18 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     check_consistent_length(y_true, y_pred, sample_weight)
 
+    if _num_samples(y_true) == 0:
```

Comment:
As discussed during the dedicated meeting, we believe that the fact that `accuracy_score` does not raise `ValueError` when called on empty arrays was an oversight:

- scikit-learn estimators always raise `ValueError` when fit on empty data,
- more importantly, they always raise `ValueError` when predicting or transforming empty data, so it's not possible to get a working pipeline that would output an empty `y_pred` array;
- many other metric functions such as `roc_auc`, `f1_score` and `mean_squared_error` already raise `ValueError`.

I think we can treat the `accuracy_score` case as a bug and make it raise `ValueError` instead (without deprecation cycle).